### PR TITLE
libsForQt5.qtwebengine: build using python3

### PR DESCRIPTION
**Pull request by dotlambda**
Motivated by https://github.com/NixOS/nixpkgs/pull/201859.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @NixOS/qt-kde

### CODE REVIEW

File: pkgs/development/libraries/qt-5/5.15/default.nix
Comment (rnhmjoj) [inline pkgs/development/libraries/qt-5/5.15/default.nix:72]: I had no idea they've had added python 3 support so late, we should thank the Arch maintainers.

File: pkgs/development/libraries/qt-5/5.12/default.nix
Comment (rnhmjoj) [inline pkgs/development/libraries/qt-5/5.12/default.nix:208]: This should be a good reason for accelerating https://github.com/NixOS/nixpkgs/issues/201765

File: pkgs/development/libraries/qt-5/5.15/default.nix
Comment (SuperSandro2000) [reply pkgs/development/libraries/qt-5/5.15/default.nix:72]: It is not proper python3 support. eg one of the checks is just replaced with return true.
Is there some upstream thing we can link? That should hopefully make it easier to keep this in the future.

File: pkgs/development/libraries/qt-5/modules/qtwebengine.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/qt-5/modules/qtwebengine.nix:60]: Not a fan of relying on packages variables being null.

Comment (dotlambda) [reply pkgs/development/libraries/qt-5/modules/qtwebengine.nix:60]: One alternative would be adding a `postPatch` argument in `qtwebengine.nix` that's appended to the existing `postPatch` string.

File: pkgs/development/libraries/qt-5/5.15/default.nix
Comment (K900) [reply pkgs/development/libraries/qt-5/5.15/default.nix:72]: Nothing upstream, the check is for the presence of a working python3 which we know we provide, so I think it's fine.

File: pkgs/development/libraries/qt-5/5.12/default.nix
Comment (K900) [reply pkgs/development/libraries/qt-5/5.12/default.nix:208]: What's going to break if we just mark it broken here?

Comment (vcunat) [reply pkgs/development/libraries/qt-5/5.12/default.nix:208]: When python2 gets marked as insecure, these will also drop anyway (by default).

File: pkgs/development/libraries/qt-5/modules/qtwebengine.nix
Comment (dotlambda) [reply pkgs/development/libraries/qt-5/modules/qtwebengine.nix:60]: @SuperSandro2000 Would you prefer that?

Comment (samueldr) [reply pkgs/development/libraries/qt-5/modules/qtwebengine.nix:60]: What's the problem with handling dependency-injected dependencies being null?

Comment (dotlambda) [reply pkgs/development/libraries/qt-5/modules/qtwebengine.nix:60]: One problem is that adding a package called catapult to Nixpkgs would break the build of e.g. `libsForQt514.qtwebengine`. Maybe I should explicitly set `catapult = null` there.

Comment (samueldr) [reply pkgs/development/libraries/qt-5/modules/qtwebengine.nix:60]: Right, forgot about this, coupled with the way we have nested set this is a problem.

Maybe somehow signal the intent in the name of the parameter? In what situations is this being set to null? I don't see that being docuemnted.

If this is for "bootstrapping" the build, a parameter for `buildWithCatapult ? true` with `false` being set when bootstrapping would explain why. Otherwise some name explaining the use-case.

If this is a python2/python3 difference, could this be handled by looking at the python interpreter to discern the reason? (And with a comment explaining why.)

Comment (dotlambda) [reply pkgs/development/libraries/qt-5/modules/qtwebengine.nix]: I decided to just add a `postPatch` argument to the expression for qtwebengine. It's the most generic solution.

**Review by FRidh — APPROVED** [review]

File: pkgs/development/libraries/qt-5/modules/qtwebengine.nix
Comment (K900) [reply pkgs/development/libraries/qt-5/modules/qtwebengine.nix]: I still think we should just rip off the band-aid now and mark it broken on pre-5.15. https://github.com/NixOS/nixpkgs/issues/201765 is basically halfway done, let's not introduce extra complexity to accomodate something that will be gone in a week anyway.

Comment (dotlambda) [reply pkgs/development/libraries/qt-5/modules/qtwebengine.nix]: It's really not a lot of extra complexity.

Comment (K900) [reply pkgs/development/libraries/qt-5/modules/qtwebengine.nix]: I'll leave it up to you.

Comment (SuperSandro2000) [reply pkgs/development/libraries/qt-5/modules/qtwebengine.nix]: > @SuperSandro2000 Would you prefer that?

I like the sound of that a lot more. :)

> What's the problem with handling dependency-injected dependencies being null?

It can lead to strange bugs and quirks like not noticing when it is being not used in *buildInputs or not being able to access attributes of it. It is just so much easier to assume a package is always defined even if unsupported on the platform.

**Pull request merged**
